windows/ReactNativeCameraCPP/CameraRotationHelper.cpp

```diff
@@ -0,0 +1,242 @@
+    {
+        m_cameraEnclosureLocation = location;
+
+        if (!IsEnclosureLocationExternal(m_cameraEnclosureLocation) && m_orientationSensor != nullptr)
+        {
+            m_sensorOrientationChanged_revoker = m_orientationSensor.OrientationChanged(winrt::auto_revoke, [ref = get_weak()](auto const& sender, auto const& args) 
+            {
+                if (auto self = ref.get()) {
+                    self->SimpleOrientationSensor_OrientationChanged(sender, args);
+                }
+            });
+        }
+        m_displayOrientationChanged_revoker = m_displayInformation.OrientationChanged(winrt::auto_revoke, [ref = get_weak()](auto const& sender, auto const& args)
+        {
+            if (auto self = ref.get()) {
+                self->DisplayInformation_OrientationChanged(sender, args);
+            }
+        });
+    }
+
+    PhotoOrientation CameraRotationHelper::GetConvertedCameraCaptureOrientation()
+    {
+        auto orientation = GetCameraCaptureOrientation();
+        return ConvertSimpleOrientationToPhotoOrientation(orientation);
+    }
+
+    SimpleOrientation CameraRotationHelper::GetCameraCaptureOrientation()
+    {
+        if (IsEnclosureLocationExternal(m_cameraEnclosureLocation))
+        {
+            // Cameras that are not attached to the device do not rotate along with it, so apply no rotation
+            return SimpleOrientation::NotRotated;
+        }
+
+        // Get the device orientation offset by the camera hardware offset
+        auto deviceOrientation = m_orientationSensor ? m_orientationSensor.GetCurrentOrientation() : SimpleOrientation::NotRotated;
+        auto result = SubtractOrientations(deviceOrientation, GetCameraOrientationRelativeToNativeOrientation());
+
+        // If the preview is being mirrored for a front-facing camera, then the rotation should be inverted
+        if (ShouldMirrorPreview())
+        {
+            result = MirrorOrientation(result);
+        }
+        return result;
+    }
+
+    SimpleOrientation CameraRotationHelper::GetCameraPreviewOrientation()
+    {
+        if (IsEnclosureLocationExternal(m_cameraEnclosureLocation))
+        {
+            // Cameras that are not attached to the device do not rotate along with it, so apply no rotation
+            return SimpleOrientation::NotRotated;
+        }
+
+        // Get the app display rotation offset by the camera hardware offset
+        auto result = ConvertDisplayOrientationToSimpleOrientation(m_displayInformation.CurrentOrientation());
+        result = SubtractOrientations(result, GetCameraOrientationRelativeToNativeOrientation());
+
+        // If the preview is being mirrored for a front-facing camera, then the rotation should be inverted
+        if (ShouldMirrorPreview())
+        {
+            result = MirrorOrientation(result);
+        }
+        return result;
+    }
+
+    int CameraRotationHelper::GetConvertedCameraPreviewOrientation()
```

Comment:
>GetConvertedCameraPreviewOrientation [](start = 30, length = 36)

This returns an int but it isn't clear from the name of the method that the int represents an angle in degrees

Reply:
Good catch, I shall rename the function (GetCameraPreviewClockwiseDegrees), it is about only few places I made changes to the sample code.